tests/core/pyspec/eth2spec/debug/random_value.py

```diff
@@ -51,65 +51,68 @@ def get_random_ssz_object(rng: Random,
     """
     if chaos:
         mode = rng.choice(list(RandomizationMode))
-    if issubclass(typ, ByteList):
-        # ByteList array
-        if mode == RandomizationMode.mode_nil_count:
-            return typ(b'')
-        elif mode == RandomizationMode.mode_max_count:
-            return typ(get_random_bytes_list(rng, min(max_bytes_length, typ.length)))
-        elif mode == RandomizationMode.mode_one_count:
-            return typ(get_random_bytes_list(rng, min(1, typ.length)))
-        elif mode == RandomizationMode.mode_zero:
-            return typ(b'\x00' * min(1, typ.length))
-        elif mode == RandomizationMode.mode_max:
-            return typ(b'\xff' * min(1, typ.length))
-        else:
-            return typ(get_random_bytes_list(rng, rng.randint(0, min(max_bytes_length, typ.length))))
-    elif issubclass(typ, ByteVector):
+    # if issubclass(typ, ByteList):
+    #     # ByteList array
+    #     if mode == RandomizationMode.mode_nil_count:
+    #         return typ(b'')
+    #     elif mode == RandomizationMode.mode_max_count:
+    #         return typ(get_random_bytes_list(rng, min(max_bytes_length, typ.limit())))
+    #     elif mode == RandomizationMode.mode_one_count:
+    #         return typ(get_random_bytes_list(rng, min(1, typ.limit())))
+    #     elif mode == RandomizationMode.mode_zero:
+    #         return typ(b'\x00' * min(1, typ.limit()))
+    #     elif mode == RandomizationMode.mode_max:
+    #         return typ(b'\xff' * min(1, typ.limit()))
+    #     else:
+    #         return typ(get_random_bytes_list(rng, rng.randint(0, min(max_bytes_length, typ.limit()))))
+    if issubclass(typ, ByteVector):
         # Sanity, don't generate absurdly big random values
         # If a client is aiming to performance-test, they should create a benchmark suite.
-        assert typ.length <= max_bytes_length
+        assert typ.type_byte_length() <= max_bytes_length
```

Comment:
Is this an alias for `vector_length` on the `ByteVector` type?

Reply:
`type_byte_length()` is common to every single view, and computes the serialized size for fixed-width types (exception if it can't be determined, for e.g. a dynamic size. That's what min/max size getters are for instead). For a vector of bytes, it's the same as the vector length.